packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -756,11 +756,12 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
       ];
       final List<String> routeParts = initialRouteName.split('/');
       if (initialRouteName.isNotEmpty) {
-        String routeName = '';
+        final StringBuffer routeName = new StringBuffer();
         for (String part in routeParts) {
-          routeName += '/$part';
-          plannedInitialRouteNames.add(routeName);
-          plannedInitialRoutes.add(_routeNamed(routeName, allowNull: true));
+          routeName.write('/$part');
+          final String current = routeName.toString();
```

Comment:
This defeats the entire point of using a StringBuffer.

Reply:
If what you want is silence the lint without annotating you can also assign `routeName = '$routeName/$part';`. Since you still need to generate all strings this is just as efficient as it originally was.

Reply:
@alexeieleusis Thanks, good call.

@a14n That suggested change would be fine by me.

As a separate note, I guess we should file a bug on the lint to have it detect `foo = '$foo...'` as well as `foo += '...'`...

Reply:
I don't think the workaround `routeName = '$routeName/$part';` works (see the test https://github.com/dart-lang/linter/blob/master/test/rules/use_string_buffers.dart#L30-L35)